src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(animal) {
+    this.name = animal;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(animal) {
+    super(animal);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && animal instanceof Herbivore) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      const animalIndex = Animal.alive.indexOf(animal);
+
+      if (animalIndex !== -1) {
+        Animal.alive.splice(animalIndex, 1);
+      }
+    }
```

Comment:
Checklist, I had the same problem, try to reverse logic, and use filter
![image](https://user-images.githubusercontent.com/99331793/174565120-13e74863-b5c3-49f0-b484-c1e7cd6b6854.png)